sdl_android/src/main/java/com/smartdevicelink/proxy/RPCStruct.java

```diff
@@ -54,21 +62,72 @@ public void deserializeJSON(JSONObject jsonObject) throws JSONException {
 			throws JSONException {
 		return JsonRPCMarshaller.deserializeJSONObject(jsonObject);
 	}
-	
+
 	public JSONObject serializeJSON() throws JSONException {
 		return JsonRPCMarshaller.serializeHashtable(store);
 	}
 	
 	@SuppressWarnings("unchecked")
-    public JSONObject serializeJSON(byte version) throws JSONException {
-		if (version > 1) {
+    public JSONObject serializeJSON(byte protocolVersion) throws JSONException {
+		if (protocolVersion > 1) {
 			String messageType = getMessageTypeName(store.keySet());
 			Hashtable<String, Object> function = (Hashtable<String, Object>) store.get(messageType);
 			Hashtable<String, Object> parameters = (Hashtable<String, Object>) function.get(RPCMessage.KEY_PARAMETERS);
 			return JsonRPCMarshaller.serializeHashtable(parameters);
 		} else return JsonRPCMarshaller.serializeHashtable(store);
 	}
 
+	/**
+	 * This method should clean the the RPC to make sure it is compliant with the spec.
+	 * <br><br><b> NOTE:</b> Super needs to be called at the END of the method
+	 *
+	 * @param rpcVersion the rpc spec version that has been negotiated. If value is null the
+	 *                   the max value of RPC spec version this library supports should be used.
+	 *  @param formatParams if true, the format method will be called on subsequent params
+	 */
+	public void format(Version rpcVersion, boolean formatParams){
+		formatRequested = true;
+		rpcSpecVersion = rpcVersion;
+		//Should override this method when breaking changes are made to the RPC spec
+		if(formatParams == true && store != null){

```

Comment:
AS keeps suggesting simplifying the condition like this
`if(formatParams && store != null)`